File: PYTHON/2715 - Dividindo os Trabalhos I.py
import sys

input = sys.stdin.readline


    
def solve(n):
    v = list(map(int,input().split()))
    
    
    vd = sum(v)
    ve = 0
    ans = abs(ve-vd)    
    
    for i in range(n):
        ve +=v[i]
        vd -= v[i]
        ans = min(ans,abs(ve-vd))
        
    print(ans)
    
   # print(ve)
   # print(vd)

while True:
    n = input()
    
    if n =='':
        break
    solve(int(n))
   # print('^^^^')
    

File: C++/2715 - Dividindo os Trabalhos I.cpp
#include <bits/stdc++.h>
using namespace std;

void solve(int n) {
    vector<int> v(n);
    for (int i = 0; i < n; ++i)
        cin >> v[i];

    vector<int> ve(n + 1, 0);
    for (int i = 0; i < n; ++i)
        ve[i + 1] = v[i] + ve[i];

    vector<int> vd(n + 1, 0);
    int j = 0;
    for (int i = n - 1; i >= 0; --i) {
        vd[j + 1] = v[i] + vd[j];
        j++;
    }

    reverse(vd.begin(), vd.end());

    int ans = INT_MAX;
    for (int i = 0; i <= n; ++i)
        ans = min(ans, abs(ve[i] - vd[i]));

    cout << ans << endl;
}

int main() {
    string line;
    while (getline(cin, line)) {
        if (line.empty()) break;
        stringstream ss(line);
        int n;
        ss >> n;
        solve(n);
    }
    return 0;
}
